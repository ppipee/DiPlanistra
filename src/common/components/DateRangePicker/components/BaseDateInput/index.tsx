import React from 'react'

import InputForm, { InputFormProps } from 'common/components/form/InputForm'
import CalendarIcon from 'common/components/icons/CalendarIcon'
import { gray } from 'common/styles/colors'

const BaseDateInput = (props: InputFormProps) => {
	return <InputForm readOnly $suffixIcon={CalendarIcon} color={gray[500]} {...props} />
}

export default BaseDateInput
