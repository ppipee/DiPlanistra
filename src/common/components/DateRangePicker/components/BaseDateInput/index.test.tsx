import React from 'react'

import { render } from '@testing-library/react'

import BaseDateInput from '.'

describe('BaseDateInput', () => {
	it('should render DateInput correctly', () => {
		const { container } = render(<BaseDateInput />)

		expect(container.firstChild).toMatchSnapshot()
	})
})
