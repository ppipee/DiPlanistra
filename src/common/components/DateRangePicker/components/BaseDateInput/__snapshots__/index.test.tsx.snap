// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDateInput should render DateInput correctly 1`] = `
<label
  class="styled__FieldContainer-kCkUmK jRkrwK"
>
  
  <label
    class="styled__InputWrapper-iKRicm ioa-drU"
  >
    <input
      class="FieldStyle__FieldStyles-fbzgYj styled__InputFieldStyles-fynRSb fShvzC MhRQv"
      color="#7D8391"
      readonly=""
    />
    <div
      class="styled__PrefixWrapper-ldLkys styled__SuffixWrapper-cEJSDX bdkLXA foDIev"
    >
      <svg
        fill="none"
        height="20"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <mask
          height="20"
          id="mask0"
          mask-type="alpha"
          maskUnits="userSpaceOnUse"
          width="20"
          x="0"
          y="0"
        >
          <rect
            fill="white"
            height="20"
            width="20"
          />
        </mask>
        <g
          mask="url(#mask0)"
        >
          <path
            clip-rule="evenodd"
            d="M17.2002 18H2.7998C2.3579 18 2 17.6416 2 17.2002V4.3999C2 3.958 2.3579 3.6001 2.7998 3.6001H6V2.7998C6 2.3579 6.3579 2 6.7998 2C7.2417 2 7.5996 2.3579 7.5996 2.7998V3.6001H12.4004V2.7998C12.4004 2.3579 12.7583 2 13.2002 2C13.6421 2 14 2.3579 14 2.7998V3.6001H17.2002C17.6421 3.6001 18 3.958 18 4.3999V17.2002C18 17.6416 17.6421 18 17.2002 18ZM3.5996 9.2002V16.4004H16.4004V9.2002H3.5996ZM6 5.1997V6C6 6.4419 6.3579 6.7998 6.7998 6.7998C7.2417 6.7998 7.5996 6.4419 7.5996 6V5.1997H12.4004V6C12.4004 6.4419 12.7583 6.7998 13.2002 6.7998C13.6421 6.7998 14 6.4419 14 6V5.1997H16.4004V7.6006H3.5996V5.1997H6ZM7.3999 13.2002V12.4004C7.3999 11.959 7.042 11.6006 6.6001 11.6006H5.7998C5.3579 11.6006 5 11.959 5 12.4004V13.2002C5 13.6416 5.3579 14 5.7998 14H6.6001C7.042 14 7.3999 13.6416 7.3999 13.2002ZM11.1997 12.4004V13.2002C11.1997 13.6416 10.8418 14 10.3999 14H9.6001C9.1582 14 8.8003 13.6416 8.8003 13.2002V12.4004C8.8003 11.959 9.1582 11.6006 9.6001 11.6006H10.3999C10.8418 11.6006 11.1997 11.959 11.1997 12.4004ZM15 13.2002V12.4004C15 11.959 14.6421 11.6006 14.2002 11.6006H13.3999C12.958 11.6006 12.6001 11.959 12.6001 12.4004V13.2002C12.6001 13.6416 12.958 14 13.3999 14H14.2002C14.6421 14 15 13.6416 15 13.2002Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </g>
      </svg>
    </div>
  </label>
</label>
`;
